src/box/applier.cc

```diff
@@ -2227,6 +2227,20 @@ applier_f(va_list ap)
 				goto reconnect;
 			} else {
 				/* Unrecoverable errors */
+				ERROR_INJECT(
+					ERRINJ_APPLIER_FAIL_DESTROY_DELAY, {
+					say_warn("applier fail delay destroy");
+					ERROR_INJECT_YIELD(
+					ERRINJ_APPLIER_FAIL_DESTROY_DELAY);
+					/*
+					 * All fibers, stopped with fiber_cancel
+					 * can end their work now.
+					 */
+					struct errinj *inj = errinj(
+						ERRINJ_APPLIER_DESTROY_DELAY,
+						ERRINJ_BOOL);
+					inj->bparam = false;
+				});
```

Comment:
This looks pretty ugly due to the requirements of `checkpatch` for indenting. But there's nothing I can do about it(

Reply:
Yep, ugly indeed.
Try this:
```c
struct errinj *inj = errinj(ERRINJ_APPLIER_FAIL_DESTROY_DELAY);
applier_log_error(...);
applier_disconnect(...);
if (inj == NULL)
        return -1;
/* all the error injection stuff goes here. */
```

Reply:
We must yield before running triggers. This one won't work. We may replace macro with explicit code. But in general it's so unimportant

Reply:
> We must yield before running triggers.

Yes, indeed. Missed that.